Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -57,6 +62,7 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         {
             return new Route(new ICallHandler[] {
                 new RecordCallSpecificationHandler(state.PendingSpecification, state.CallSpecificationFactory, state.CallActions)
+                , new ClearLastTrackedCallHandler(state.LastCallTracker)
```

Comment:
Ok, agree, even if `GetCallSpec` has the check it's better to clear it explicitly  :)